### ISSUE TITLE
[2.14.0][Standalone] ResourceDescriptions are not bound in ResourceDescriptionsProvider

### ISSUE THREAD

**Issue by kutschkem** [eclipse/xtext-core#827]
I have issues with cross-referencing between two grammars. In an eclipse context, everything works as expected, but in Standalone mode, the cross-references fail to be resolved.

Here is how I create ResourceSets in the eclipse context:

	@Inject
	IResourceSetProvider rsp;

	@Override
	protected XtextResourceSet getResourceSet(URI uri) {
		return (XtextResourceSet) rsp.get(getProject(uri));
	}

	/**
	 * Returns project for a corresponding file.
	 */
	private IProject getProject(URI uri) {
		IProject project = ResourcesPlugin.getWorkspace().getRoot().getFile(new Path(uri.toPlatformString(true)))
				.getProject();
		return project;
	}

And here is how I do it in standalone mode:

    @Inject
    private Provider<XtextResourceSet> resourceSetProvider;

	protected XtextResourceSet getResourceSet(URI uri) {
		XtextResourceSet set = resourceSetProvider.get();
		return set;
	}

Creating the resource is the same in both cases, but for the eclipse context the cross-references are correctly found, but not in standalone mode:

 ```
 public Resource loadResource(URI uri) throws IOException {
    XtextResourceSet set = getResourceSet(uri);
    set.addLoadOption(XtextResource.OPTION_RESOLVE_ALL, Boolean.TRUE);
    
    // load root resource
    Resource resource = set.getResource(uri, true);
      resource.load(Collections.EMPTY_MAP);
      return resource;

  }

```

The issue I have identified is that in Standalone mode, the ResourceDescriptionProvider does not get injected with ResourceDescriptions, while it does get injected in the Eclipse context. In bot cases, I use the RuntimeModule, so there should be a binding for ResourceDescriptions, but nevertheless, it does not get injected.

The result is this stack trace:

```
java.lang.NullPointerException
	at org.eclipse.xtext.resource.impl.ResourceDescriptionsProvider.createResourceDescriptions(ResourceDescriptionsProvider.java:159)
	at org.eclipse.xtext.resource.impl.ResourceDescriptionsProvider.getResourceDescriptions(ResourceDescriptionsProvider.java:113)
	at org.eclipse.xtext.scoping.impl.AbstractGlobalScopeProvider.getResourceDescriptions(AbstractGlobalScopeProvider.java:37)
	at org.eclipse.xtext.scoping.impl.ImportUriGlobalScopeProvider.getResourceDescriptions(ImportUriGlobalScopeProvider.java:66)
	at org.eclipse.xtext.scoping.impl.ImportUriGlobalScopeProvider.getScope(ImportUriGlobalScopeProvider.java:75)
	at org.eclipse.xtext.scoping.impl.AbstractGlobalScopeProvider.getScope(AbstractGlobalScopeProvider.java:50)
	at org.eclipse.xtext.scoping.impl.AbstractGlobalScopeDelegatingScopeProvider.getGlobalScope(AbstractGlobalScopeDelegatingScopeProvider.java:46)
	at org.eclipse.xtext.scoping.impl.AbstractGlobalScopeDelegatingScopeProvider.getGlobalScope(AbstractGlobalScopeDelegatingScopeProvider.java:42)
	at org.eclipse.xtext.scoping.impl.SimpleLocalScopeProvider.getScope(SimpleLocalScopeProvider.java:62)
	at org.eclipse.xtext.linking.impl.DefaultLinkingService.getScope(DefaultLinkingService.java:58)
	at org.eclipse.xtext.linking.impl.DefaultLinkingService.getLinkedObjects(DefaultLinkingService.java:107)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:247)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:222)
	at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.getEObject(ResourceSetImpl.java:223)
	at org.eclipse.emf.ecore.util.EcoreUtil.resolve(EcoreUtil.java:205)
	at org.eclipse.emf.ecore.util.EcoreUtil.resolve(EcoreUtil.java:261)
	at org.eclipse.emf.ecore.impl.BasicEObjectImpl.eResolveProxy(BasicEObjectImpl.java:1477)
```

**Comment by cdietrich** [comment]
How do you create the injector.
Iproject is not a stand-alone thing

**Comment by kutschkem** [comment]
Injector globalStateInjector = Guice.createInjector(
          new IniConfigModule(iniConfig),
          new LoggingModule(LoggingBindings.CL, null)
          );
      
      ProtobufStandaloneSetup.doSetup(globalStateInjector);    //Setup language 1
      globalStateInjector = YaafStandaloneSetup.doSetup(globalStateInjector);  //Setup language 2

**Comment by cdietrich** [comment]
is that the project where createChildInjector is used all over the place in some internals? dont use createChildinjectors. create injectos only on top level.

**Comment by kutschkem** [comment]
Yes that is right.

**Comment by kutschkem** [comment]
Removing the childinjector from the StandaloneSetup helped, for some reason. The modules that end up in the injector are the IniConfigModule, LoggingModule, which both just bind a handful of inctances, and the YaafRuntimeModule.

This was then used for further childInjector creation, which did in fact in the past cause problems with JIT-bindings, but I worked around that by providing an explicit binding for the offending class.

Anyway, removing the "further child injector creation" did not help, only when I removed the childInjector creation from the StandaloneSetup, my errors went away.
